examples/lstm_stateful.py

```diff
@@ -0,0 +1,204 @@
+'''Example script showing how to use stateful RNNs
+to model long sequences efficiently.
+Depending on the parameters, either only the stateful LSTM converges,
+or both the stateful and stateless LSTM converge.
+'''
+from __future__ import print_function
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+# -------------------
+# EDITABLE PARAMETERS
+# -------------------
+
+# timesteps to use in the output averaging of the input
+# e.g. use 2 for 2-point average
+tsteps = 2
+
+# number of elements ahead that are used to make the LSTM prediction
+# when lahead >= tsteps, the stateless LSTM can converge
+# similar to the stateful LSTM
+lahead = 1
+
+# length of input
+input_len = 1000
+
+# training parameters
+batch_size = 1
+epochs = 10
+
+# ------------
+# MAIN PROGRAM
+# ------------
+
+print("*" * 33)
+if lahead >= tsteps:
+    print("STATELESS LSTM WILL ALSO CONVERGE")
+else:
+    print("STATELESS LSTM WILL NOT  CONVERGE")
+print("*" * 33)
+
+# add a few points to account for the
+# nan-values that will be dropped in preprocessing
```

Comment:
What does this mean?

Reply:
I extended the comment to explain as follows:

Since the output is a moving average of the input,
the first few points of output will be NaN
and will be dropped from the generated data
before training the LSTM.
Also, when lahead > 1, the preprocessing step later of "rolling window view"
will also cause some points to be lost.
For aesthetic reasons,
in order to maintain generated data length = input_len after pre-processing,
add a few points to account for the values that will be lost.